lib/chrome.js

```diff
@@ -290,20 +298,30 @@ class ChromeTab {
   }
 
   onMessageAdded({ message }) {
-    console.log(`[${this.id}]`, message.text)
-    if (message.text == 'Zen.idle' && this.state == 'loading') {
-      this.becomeIdle()
-    }
+    const text = message.text
+    const register = ((name, cb) => {
```

Comment:
No changes needed. 

Did you consider a switch instead of a register helper? 

```
switch(true){
     case message.text.startsWith('Zen.hotReload') && this.state === 'hotReload':
     case message.text.startsWith('Zen.idle') && this.state === 'loading':
         this.becomeIdle()
         break;
     case message.text.startsWith('Zen.results') && this.state === 'running':
         let msg = JSON.parse(message.text.slice(12))
         this.finishTest(msg)
         this.becomeIdle()
         break;
     case message.text.startsWith('Zen.resizeWindow'):
         this.resizeWindow()
         break;
}

```

Reply:
Maybe 🤷 

The big reason I wanted to use the register helper was because anything sending a message would require the JSON.parse section and then you would need to copy that every time you made a new handler.

I expect a larger refactor at some point, I really want all this message passing to be much more structured and typed.